### PR TITLE
clean up proxy repeating code

### PR DESCRIPTION
**Pull request by ikaven1024**
Signed-off-by: yingjinhui <yingjinhui@didiglobal.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
merge some repeating code

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/registry/cluster/storage/proxy_test.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+			}
+			h, err := r.Connect(req.Context(), tt.args.id, tt.args.options, utiltest.NewResponder(resp))
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Connect() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			if err != nil {
+				return
+			}
```
Comment (cmicat) [inline pkg/registry/cluster/storage/proxy_test.go:135]: why check err again with different logic?

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~
/lgtm

**Review by RainbowMango — APPROVED** [review]
/approve

File: pkg/registry/cluster/storage/proxy_test.go (hunk 1 above)
Comment (XiShanYongYe-Chang) [reply pkg/registry/cluster/storage/proxy_test.go:135]: I think the reason is that when the error is not nill, there is no need to execute subsequent logic.

Comment (ikaven1024) [reply pkg/registry/cluster/storage/proxy_test.go:135]: yes

**Pull request merged**